### DISCUSSION TITLE
App format for Linux users

### DISCUSSION THREAD

**Discussion by louisgab** [spatie/ray#242] [Ideas]
@freekmurze I was wondering if, by any chance, you could consider building an other format for Linux users ?

I mean AppImage is fine, but it behaves likes a "portable app", not something you really install and can easily launch.
In fact, we have to go each time in the folder where it is and then execute it ... 
There is no way to show it in an application list menu, or pin it to a dock bar, neither fuzzy find it in something like ulauncher (kind of spotlight/alfred)

From a statistic point of view, at least 66.2% of linux users can go for a .deb format and 16.6% for a .rpm format
(according to [https://w3techs.com/technologies/details/os-linux](https://w3techs.com/technologies/details/os-linux))
But it seems their is a new universal way, with the snap format so if you have to choose only one format it would be snap!
Maybe that's why [Slack](https://slack.com/intl/fr-fr/downloads/linux), for example, choose to provide all three versions.

I guess the app is built with electronjs, maybe using electron-builder, so it can be a matter of changing the default `target = AppImage` to `snap`

**Comment by AdrianMrn** [comment]
@pascalchevrel Just to get a second opinion, is this something that would benefit you as well?

**Comment by pascalchevrel** [comment]
You currently have 2 options for universal packaging on Linux which are snap (sponsored by Canonical) and flatpak (sponsored by Redhat). I don't have a strong opinion on either, you should just know that they both provide some security sandboxing features to install the binary in the home folder so you might want to check that it doesn't block communicating with the local server and opening files in the IDE. 

Personally I would be happy with either a .deb or a snap as they are the common formats on Ubuntu. Having a snap package would probably allow better integration with Gnome out of the box (Gnome launcher, management of the binary via the package manager) and snaps can be installed on other distros not based on .deb packages. 

The AppImage format is indeed rather uncommon and the fact that each binary update changes the exe filename makes it hard to create a .desktop file and maintain your own launcher (I don't want to update the launcher file for every update of the app), personally I gave up and launch Ray from a terminal which is indeed a  bit weird for a visual app.

**Reply by patinthehat** [reply]
@pascalchevrel See [this page](https://spatie.be/docs/ray/v1/environment-specific-configuration/linux) of the docs for a solution to the AppImage update rename issue.

**Comment by AdrianMrn** [comment]
Thanks so much for the detailed information, both of you! We'll have a look at this in the coming weeks, and I'll update here once we've made a decision and implemented it 👍

**Reply by louisgab** [reply]
@AdrianMrn so Adrian, any update ?

**Reply by AdrianMrn** [reply]
Thanks for the reminder, I'll bump this up in our internal issues board. It got lost in the various performance improvements and additional features we added over the last couple of weeks 😬
